numpy/lib/arraysetops.py

```diff
@@ -254,42 +316,94 @@ def reshape_uniq(uniq):
         uniq = np.swapaxes(uniq, 0, axis)
         return uniq
 
+    consolidated_base = np.array_view_root(consolidated)
+    orig_base = np.array_view_root(orig_ar)
+    if not indirect_sort and orig_base is not consolidated_base:
+        # We're operating on a copy, so we might as well
+        # do an in-place sort on that copy
+        sort_inplace = True
```

Comment:
Why not:
```
if return_index or return_inverse:
    sort_inplace = False
else if orig_base is not consolidated_base:
    sort_inplace = True
```
and eliminate the stuff below.

AT the very least, this would be much clearer as an `else` on the `if indirect_sort`

Reply:
This is pretty much that, except the first part is at the beginning of the function so the 1d case also gets the `sort_inplace = False` bit.